lib/lib_ssl/tls_mini/src/WiFiClientSecureLightBearSSL.h

```diff
@@ -131,6 +131,9 @@ class WiFiClientSecure_light : public WiFiClient {
     }
 
   private:
+    unsigned long lastOutActivity;
+    unsigned long lastInActivity;
+    int _loopTimeout=15;

```

Comment:
Isn't 14 seconds too long? I haven't seen such issue yet, but 5 seconds seems to me already long enough considering that the whole Tasmota is already blocked. 

Maybe we can also reuse the http timeout, hence we need a setter function to change the timeout

Reply:
Actually, I just copied this 15 seconds from the Arduino implementation of `WiFiClientSecureBearSSL`, https://github.com/esp8266/Arduino/blob/c96dc26792905dc43280587fcdf5ae5af024b286/libraries/ESP8266WiFi/src/WiFiClientSecureBearSSL.cpp#L249
I thought,  we need that much time in a case where internet connection is slow. In this period whole Tasmota get blocked and can't access the web server of the tasmota. As we have the timeout increment for the mqtt connection. I thought that not be a much problem. Can't we `yield` by default at this `_run_until` loop?. This is the flow after the fix.

```
11:05:59.563 MQT: Authenticating with an Azure IoT Hub Token
11:05:59.972 MQT: TLS connection error: -1000
11:05:59.975 MQT: Connect failed to xxx-xxxx-xxxx-xxx-xx.azure-devices.net:8883, rc -2. Retry in 80 sec
11:07:20.656 MQT: Attempting connection...
11:07:21.248 MQT: Authenticating with an Azure IoT Hub Token
11:07:21.687 MQT: TLS connection error: -1000
11:07:21.690 MQT: Connect failed to xxx-xx-xxx-xxx-xx.azure-devices.net:8883, rc -2. Retry in 90 sec
11:08:52.488 MQT: Attempting connection...
11:08:52.621 MQT: Authenticating with an Azure IoT Hub Token
11:08:52.927 MQT: TLS connection error: -1000
11:08:52.929 MQT: Connect failed to xxx-xx-xxx-xxx-xx.azure-devices.net:8883, rc -2. Retry in 100 sec
```

Also can you please point out this  `http timeout` method that you mentioned.

Reply:
I set the timeout to 5 seconds and checked with simulating the network connection speed. It worked without having  an issue. @s-hadinger

Reply:
Thanks, let's start with 5 seconds. We can add it as customizable if needed.